R/type-list-of.R

```diff
@@ -62,7 +62,10 @@ new_list_of <- function(x = list(), ptype = logical(), ..., class = character())
   stopifnot(is.list(x))
   stopifnot(vec_size(ptype) == 0)
 
-  new_vctr(x, ..., ptype = ptype, class = c(class, "vctrs_list_of"))
+  # Append 'list' class after 'vctrs_vctr'
+  out <- new_vctr(x, ..., ptype = ptype, class = c(class, "vctrs_list_of"))
+  class(out) <- c(class(out), "list")
```

Comment:
I think you can just inline `list` into the class list — the point is not to preserve the existing class, but to lift the base type (which becomes the implicit class) into the subclass. 

I'm still not sure this is correct but I'm thinking about it.

Reply:
Such that the resulting `list_of` class will be `c("vctrs_list_of", "list", "vctrs_vctr")` ?

Reply:
I think `"list"` should be last because we can't superclass foreign classes.

Reply:
Right, that's why I implemented it this way. So keep the PR as is?

Reply:
I wonder if we need another argument to `new_vctr()`? Like `new_vctr(x, ..., ptype = ptype, class = c(class, "vctrs_list_of"), preserve_implict_class = TRUE)` (but a better name)

Reply:
oh yes that's occasionally useful.

```r
explicit_base_class = TRUE
base_class = TRUE
subclass_base_type = TRUE
```

Maybe the first one?

Reply:
Or `explicit_type`? Otherwise, I like `explicit_base_class` best of the options so far.

Reply:
Another option would be `include_type_in_class` (that's too long, but might suggest other ideas)

Reply:
`explicit_typeof` might also work, depending on how we'd extract the base type

Reply:
Should I try to add something like this to the PR? Anyone with strong opinions want to take it from here?

Reply:
No worries @jeroen, I can finish the PR when I get back to work on vctrs.

Reply:
How about `extends_type`?